src/Matrix/AMatrixDense.cpp

```diff
@@ -92,6 +92,12 @@ double AMatrixDense::_getValueByRank(int irank) const
   return *(_eigenMatrix.data() + irank);
 }
 
+std::span<const double> AMatrixDense::getColumnPtr(int icol) const
+{
+  const auto a = _eigenMatrix.col(icol);
+  int n = getNRows();
+  return std::span<const double>(a.data(),n);
```

Comment:
I hope `a` is a non-owning pointer inside `_eigenMatrix`, otherwise this method might trigger a use-after-free.